incubator/model-core/src/main/java/io/aklivity/zilla/runtime/model/core/internal/IntegerFormat.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.model.core.internal;
+
+import static io.aklivity.zilla.runtime.engine.model.ValidatorHandler.FLAGS_FIN;
+import static io.aklivity.zilla.runtime.engine.model.ValidatorHandler.FLAGS_INIT;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+
+public enum IntegerFormat
+{
+    TEXT
+    {
+        private boolean number;
+        private boolean sign;
+        int decimalPlaces;
+        private int value;
```

Comment:
When we use enum as a strategy object, the implementation needs to be stateless because the same instance of the enum constant is reused by all callers, so any state stored on the instance of the enum constant is also shared by all callers, causing a collision either across threads or due to overlapping fragmented validate calls even on the same thread.

Suggest moving the validation to a different class that can have state, and keeping just the remaining necessary behavioral parts here in the enum that can perform the checks needed by the stateful validator.